plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/CollationAwareQueryBuilder.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.postgresql;
+
+import io.trino.plugin.jdbc.DefaultQueryBuilder;
+import io.trino.plugin.jdbc.JdbcClient;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcJoinCondition;
+import io.trino.spi.type.CharType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+
+import java.util.stream.Stream;
+
+import static java.lang.String.format;
+
+public class CollationAwareQueryBuilder
+        extends DefaultQueryBuilder
+{
+    @Override
+    protected String formatJoinCondition(JdbcClient client, String leftRelationAlias, String rightRelationAlias, JdbcJoinCondition condition)
+    {
+        boolean needsCollation = Stream.of(condition.getLeftColumn(), condition.getRightColumn())
+                .map(JdbcColumnHandle::getColumnType)
+                .anyMatch(CollationAwareQueryBuilder::isCharType);
+
+        if (needsCollation) {
+            return format(
+                    "%s.%s COLLATE \"C\" %s %s.%s COLLATE \"C\"",
+                    leftRelationAlias,
+                    buildJoinColumn(client, condition.getLeftColumn()),
+                    condition.getOperator().getValue(),
```

Comment:
Could this cause a performance regression? A query that previously used an index to join two tables now might not because of the collation

Reply:
Should have continued reading the PR, I see we only gate pushdown of less than/greater than operators behind a config flag but not equality.

This means will add the extra `collate c` bit into every pushed down join condition. For the specific case of equality joins this shouldn't be a problem right (why would collation be necessary for an equality join)? Might want to double check though

Reply:
Correct. I raised this concern in https://github.com/trinodb/trino/pull/9746 and hence the feature was gated behind a config flag.

Maybe we can be smarter and only apply the collation for inequalities and ranges. (However I think it's possible for collations to affect equality comparisons as well but I'm not sure).

Reply:
This is also the time when we can toggle that to `true` by default since it should no longer cause a performance regression in the equality case.

Reply:
I've moved collation logic to CollationAwareQueryBuilder